### PR TITLE
Add an integration test for Delta Iceberg Converter

### PR DESCRIPTION
**Pull request by jackierwzhang**
## Description
Added an integration test. Run as follows:
```
python run-integration-tests.py --run-iceberg-integration-tests --use-local [--iceberg-spark-version 3.3] [--iceberg-lib-version 1.0.0]
```

## How was this patch tested?
The run succeeds and outputs as expected.

## Does this PR introduce _any_ user-facing changes?
No.

### CODE REVIEW

File: delta-iceberg/integration_tests/iceberg_converter.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from pyspark.sql.functions import col
+from delta.tables import DeltaTable
+import shutil
+import random
+
+testRoot = "/tmp/delta-iceberg-converter/"
+warehousePath = testRoot + "iceberg_tables"
+shutil.rmtree(testRoot, ignore_errors=True)
+
+# Enable SQL commands and Update/Delete/Merge for the current spark session.
```
Comment (vkorukanti) [inline delta-iceberg/integration_tests/iceberg_converter.py:27]: update the comment. we are not using any Update/Delete/Merge. Add a comment about iceberg catalog.

File: delta-iceberg/integration_tests/iceberg_converter.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+print("Reading the final Delta table...")
+spark.read.format("delta").load(tablePath).show()
+
+print("Create an external catalog table using Delta...")
+spark.sql("CREATE TABLE converted_delta_table USING delta LOCATION '{}'".format(tablePath))
+
+print("Read from the catalog table...")
+spark.read.table("converted_delta_table").show()
+
+# cleanup
```
Comment (vkorukanti) [inline delta-iceberg/integration_tests/iceberg_converter.py:67]: use try..finally

File: run-integration-tests.py [hunk 3]
```diff
@@ -196,6 +196,43 @@ def run_s3_log_store_util_integration_tests():
...
+    if use_local:
+        run_cmd(["build/sbt", "publishM2"])
+
+    test_dir = path.join(root_dir, path.join("delta-iceberg", "integration_tests"))
+    files_to_skip = {"using_with_pip.py", "missing_delta_storage_jar.py", "image_storage.py"}
+
+    test_files = [path.join(test_dir, f) for f in os.listdir(test_dir)
+                  if path.isfile(path.join(test_dir, f)) and
+                  f.endswith(".py") and not f.startswith("_") and
+                  f not in files_to_skip]
```
Comment (vkorukanti) [inline run-integration-tests.py:211]: given there is only one test, just use that directly than listing everything and excluding.

**Review by vkorukanti — APPROVED** [review]
minor comments, lgtm

File: run-integration-tests.py (hunk 3 above)
Comment (jackierwzhang) [reply run-integration-tests.py:211]: I think this allows us to add other iceberg integration suites in the same directory as well, so it's ok to have this way?

Comment (vkorukanti) [reply run-integration-tests.py:211]: When that happens, we can add the test to `test_files` list here.

Comment (jackierwzhang) [reply run-integration-tests.py:211]: alright, i don't have strong opinion on this

**Pull request closed without merging**